lib/mysql.js

```diff
@@ -18,14 +18,48 @@ var debug = require('debug')('loopback:connector:mysql');
  * @param {Function} [callback] The callback function
  */
 exports.initialize = function initializeDataSource(dataSource, callback) {
-  if (!mysql) {
-    return;
-  }
+  dataSource.driver = mysql; // Provide access to the native driver
+  dataSource.connector = new MySQL(dataSource.settings);
+  dataSource.connector.dataSource = dataSource;
+
+  defineMySQLTypes(dataSource);
+
+  dataSource.EnumFactory = EnumFactory; // factory for Enums. Note that currently Enums can not be registered.
+
+  process.nextTick(function () {
+    callback && callback();
+  });
+};
+
+exports.MySQL = MySQL;
+
+function defineMySQLTypes(dataSource) {
+  var modelBuilder = dataSource.modelBuilder;
+  var defineType = modelBuilder.defineValueType ?
+    // loopback-datasource-juggler 2.x
+    modelBuilder.defineValueType.bind(modelBuilder) :
+    // loopback-datasource-juggler 1.x
+    modelBuilder.constructor.registerType.bind(modelBuilder.constructor);
+
+  // The Point type is inherited from jugglingdb mysql adapter.
+  // LoopBack uses GeoPoint instead.
+  // The Point type can be removed at some point in the future.
+  defineType(function Point() {
+  });
```

Comment:
An idea to consider: drop deprecated stuff like `Point` and support for loopback-datasource-juggler 1.x, and release the changes as a major version.